### PR TITLE
2791: Add handling for double clicks on Wicket ajax components

### PR DESCRIPTION
**Pull request by payten**
Delivers #2791, #1915.

Introduce `GbAjaxButton` and `GbAjaxLink` to offer some kind of double-click handling for each.

- `GbAjaxButton` disables the button on click and re-enables upon complete.
- `GbAjaxLink` introduces a link channel so if another request of the same type comes through it is ignored (http://apache-wicket.1842946.n4.nabble.com/How-to-prevent-a-concurrent-click-on-ajax-links-AjaxFallbackLink-td4660226.html)
- Also disable all inputs on a modal as it is closing, just in case it takes a moment for the modal to close.

### CODE REVIEW

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/component/GbAjaxLink.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	}
+
+	public GbAjaxLink(String id, IModel<T> model) {
+		super(id, model);
+	}
+
+	@Override
+	protected void updateAjaxAttributes(AjaxRequestAttributes attributes) {
+		super.updateAjaxAttributes(attributes);
+		attributes.setChannel(new AjaxChannel("blocking", AjaxChannel.Type.ACTIVE));
```
Comment (steveswinsburg) [inline gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/component/GbAjaxLink.java]: they thought of everything!

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/component/GbAjaxLink.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * A GradebookNG implementation of Wicket's AjaxLink that
+ * disables the link on click to reduce the likelihood of
+ * double-clicks resulting in double actions.
+ */
+abstract public class GbAjaxLink<T> extends AjaxLink {
+
+	public GbAjaxLink(String id) {
+		this(id, (IModel)null);
```
Comment (steveswinsburg) [inline gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/component/GbAjaxLink.java]: can this just be `this(id)` rather than a cast for a null?

**Comment by steveswinsburg** [comment]
Nice solution Payten! ♨️ 🎉

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/component/GbAjaxLink.java (hunk 2 above)
Comment (payten) [inline gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/component/GbAjaxLink.java]: Hm yeah.. I copied from the AjaxLink class, but you're totally right, the cast/null is not required.

**Pull request merged**